segmentstore/server/host/src/main/java/io/pravega/service/server/host/stat/AutoScaleProcessor.java

```diff
@@ -185,18 +186,29 @@ private void triggerScaleDown(String streamSegmentName, boolean silent) {
 
     private CompletableFuture<Void> writeRequest(ScaleEvent event) {
         CompletableFuture<Void> result = new CompletableFuture<>();
+        CompletableFuture<Void> writeComplete = new CompletableFuture<>();
         try {
-            CompletableFuture.runAsync(() -> {
+            AckFuture ackFuture = writer.get().writeEvent(event.getKey(), event);
+            ackFuture.addListener(() -> writeComplete.complete(null), executor);
+            writeComplete.thenAcceptAsync((Void v) -> {
                 try {
-                    writer.get().writeEvent(event.getKey(), event).get();
-                    result.complete(null);
-                } catch (InterruptedException | ExecutionException e) {
-                    log.error("error sending request to requeststream {}", e);
+                    ackFuture.get();
+                } catch (ExecutionException | InterruptedException e) {
+                    log.error("Sending scale event failed {}/{}/{}", event.getScope(), event.getStream(), event.getSegmentNumber());
+                    result.completeExceptionally(ExceptionHelpers.getRealException(e));
+                }
+            }).whenComplete((r, e) -> {
+                if (e != null) {
+                    log.error("Sending scale event failed {}/{}/{}", event.getScope(), event.getStream(), event.getSegmentNumber());
                     result.completeExceptionally(e);
+                } else {
+                    log.info("Sending scale event succeeded {}/{}/{}", event.getScope(), event.getStream(), event.getSegmentNumber());
+
+                    result.complete(null);
                 }
-            }, executor);
-        } catch (RejectedExecutionException e) {
-            log.error("our executor queue is full. failed to post scale event for {}/{}/{}", event.getScope(), event.getStream(), event.getSegmentNumber());
+            });
+        } catch (Exception e) {
```

Comment:
almost everything in this catch block is async it would be a lot more clear what it was intended to cover if it were scoped down.